packages/@aws-cdk/aws-apprunner-alpha/lib/auto-scaling-configuration.ts

```diff
@@ -0,0 +1,194 @@
+  /**
+   * Imports an App Runner Auto Scaling Configuration from attributes
+   */
+  public static fromAutoScalingConfigurationAttributes(scope: Construct, id: string,
+    attrs: AutoScalingConfigurationAttributes): IAutoScalingConfiguration {
+    const autoScalingConfigurationArn = attrs.autoScalingConfigurationArn;
+    const autoScalingConfigurationName = attrs.autoScalingConfigurationName;
+    const autoScalingConfigurationRevision = attrs.autoScalingConfigurationRevision;
+
+    class Import extends cdk.Resource {
+      public readonly autoScalingConfigurationArn = autoScalingConfigurationArn
+      public readonly autoScalingConfigurationName = autoScalingConfigurationName
+      public readonly autoScalingConfigurationRevision = autoScalingConfigurationRevision
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * Imports an App Runner Auto Scaling Configuration from its ARN
+   */
+  public static fromArn(scope: Construct, id: string, autoScalingConfigurationArn: string): IAutoScalingConfiguration {
+    const arn = cdk.Stack.of(scope).splitArn(autoScalingConfigurationArn, cdk.ArnFormat.SLASH_RESOURCE_NAME);
+
+    const resourceParts = arn.resourceName?.split('/');
+
+    if (!resourceParts || resourceParts.length < 3) {
+      throw new Error(`Unexpected ARN format: ${autoScalingConfigurationArn}`);
+    }
+
+    const autoScalingConfigurationName = resourceParts[0];
+    const autoScalingConfigurationRevision = parseInt(resourceParts[1]);
+
+    return AutoScalingConfiguration.fromAutoScalingConfigurationAttributes(scope, id, {
+      autoScalingConfigurationArn,
+      autoScalingConfigurationName,
+      autoScalingConfigurationRevision,
+    });
+  }
+
+  /**
+    * The ARN of the Auto Scaling Configuration.
+    * @attribute
+    */
+  readonly autoScalingConfigurationArn: string;
+
+  /**
+    * The name of the Auto Scaling Configuration.
+    * @attribute
+    */
+  readonly autoScalingConfigurationName: string;
+
+  /**
+   * The revision of the Auto Scaling Configuration.
+   * @attribute
+   */
+  readonly autoScalingConfigurationRevision: number;
+
+  public constructor(scope: Construct, id: string, props: AutoScalingConfigurationProps = {}) {
+    super(scope, id, {
+      physicalName: props.autoScalingConfigurationName,
+    });
+
+    if (props.autoScalingConfigurationName !== undefined) {
+      if (props.autoScalingConfigurationName.length < 4 || props.autoScalingConfigurationName.length > 32) {
+        throw new Error(`autoScalingConfigurationName must be between 4 and 32 characters long, but it has ${props.autoScalingConfigurationName.length} characters.`);
+      }
+
+      if (!/^[A-Za-z0-9][A-Za-z0-9\-_]{3,31}$/.test(props.autoScalingConfigurationName)) {
+        throw new Error(`autoScalingConfigurationName ${props.autoScalingConfigurationName} must start with a letter or number, and can contain only letters, numbers, hyphens, and underscores.`);
+      }
+    }
```

Comment:
```suggestion
    if (
      props.autoScalingConfigurationName !== undefined && 
      !cdk.Token.isUnresolved(props.autoScalingConfigurationName) &&
      !/^[A-Za-z0-9][A-Za-z0-9\-_]{3,31}$/.test(props.autoScalingConfigurationName)
    ) {
      throw new Error(`autoScalingConfigurationName must match the ^[A-Za-z0-9][A-Za-z0-9\-_]{3,31}$ pattern, got ${props.autoScalingConfigurationName}`);
    }
```
We could simplify as the check on the length is redundant.